### ISSUE TITLE
[csrng] State machine error coverage

### ISSUE THREAD

**Issue by marnovandermaas** [lowRISC/opentitan#16388]
There are 5 state machines inside CSRNG, 2 of those have no to very little coverage of the error state transitions. I have found a root cause for this.

For `csrng_cmd_stage` the error state is only reached when `u_prim_count_cmd_gen_cntr` outputs an error. Similarly in `main_sm` the error states are only reached when any of the redundant counter errors are raised. In `main_sm` the idle to error transition is the only one that is tested, so we need the tests to cause these counter errors while it is in non-idle states.

Suggested solutions:

- [x] Write a test that continuously generates random bits and toggles `cs_aes_halt_i` at random intervals. (This is already done bye the `csrng_cmds` test.)
- [x] While sending random commands to random apps in the CSRNG (to get diverse states in the command stage as well as the main state machine, cause counter mismatches in `u_prim_count_cmd_gen_cntr`, `u_prim_count_ctr_drbg` and `u_prim_count_ctr_drbg` at random intervals.

This issue contributes to: https://github.com/lowRISC/opentitan/issues/15748

Estimate before 2022-11-17 4
> estimate 8
> remaining 2022-11-17 4
> remaining 2022-11-18 1

**Comment by marnovandermaas** [comment]
The solution would involve extending the `cmd_gen_cnt_err` case in `csrng_err_vseq.sv` to also send commands to the different apps so that they get into interesting states:
https://github.com/lowRISC/opentitan/blob/5050d25832b54ad79f07a48fd26bd2b84bbb8f72/hw/ip/csrng/dv/env/seq_lib/csrng_err_vseq.sv#L125

**Comment by marnovandermaas** [comment]
During the V2S meeting, it was suggested to that we also make a change to the FSM code similar to EDN: https://github.com/lowRISC/opentitan/pull/16307

**Comment by vogelpi** [comment]
Thanks for looking into this and sharing your findings @marnovandermaas . The plan looks good to me. I think it would make the `csrng_err` test much stronger if in parallel commands would be sent to the different apps in general (also when testing other things than the `cmd_gen_cnt_err`). That's also what @rasmus-madsen did for AES.

**Comment by marnovandermaas** [comment]
PR: https://github.com/lowRISC/opentitan/pull/16397 just got merged, this comment is to update the effort here.

> spent 4
> remaining 2022-11-17 4

**Comment by vogelpi** [comment]
Update: @marnovandermaas and I discussed that there is an easy way to keep running background CSRNG commands while injecting errors. This is still on our list.

**Comment by marnovandermaas** [comment]
I think this PR should be good enough for now: https://github.com/lowRISC/opentitan/pull/16473

Unless someone objects, I think we can close this issue once that PR is merged.

> spent 3
> remaining 2022-11-18 1
